bin/__gaffer.py

```diff
@@ -41,21 +41,37 @@
 # as it ensures the correct process environment is set up prior to launch.
 
 import os
+import re
 import sys
 import signal
 import warnings
+import setproctitle
 
 # Get rid of the annoying signal handler which turns Ctrl-C into a KeyboardInterrupt exception
 signal.signal( signal.SIGINT, signal.SIG_DFL )
 
+# Rename our process from `python __gaffer.py ...` to `gaffer ...`.
+# The details of the recovery and clobbering of argv and argc are
+# heinous, so we outsource it to `setproctitle`.
+setproctitle.setproctitle(
+	re.sub( "^python .*?__gaffer.py", "gaffer", setproctitle.getproctitle(), count = 1 )
```

Comment:
Not sure if this is relevant or not, might we need `python[0-9]?` for future compatibility?

Reply:
I don't think that's necessary - the `bin/gaffer` wrapper always runs `python`, not `python2` or `python3`.

Reply:
And I sincerely hope the Python folks will think twice before making a Python 4. Unless it gets rid of the GIL.